MetaMorpheus/EngineLayer/SpectralLibrarySearch/LibrarySpectrum.cs

```diff
@@ -39,6 +42,34 @@ public LibrarySpectrum(string sequence, double precursorMz, int chargeState, Lis
             Array.Sort(XArray, YArray);
         }
 
+        /// <summary>
+        /// This function enables the spectrum angle to be computed between an individual experimental spectrum and the loaded library spectrum within MetaDraw
+        /// </summary>
+        /// <param name="librarySpectrum"></param>
+        /// <returns></returns>
+        public string CalculateSpectralAngleOnTheFly(List<MatchedFragmentIon> spectrumMatchFragments)
```

Comment:
Does returning a string make sense here?

Reply:
there is cases for an againts. i guess you could return NaN when there is no angle. but I am just returning the displayed string. it has to be converted to display string and formated at some point.